### ISSUE TITLE
DefaultBigText  Style problem

### ISSUE THREAD

**Issue by redtoolskobe** [pterm/pterm#255]
macos ZSH_THEME="powerlevel9k/powerlevel9k"   Hack  nerd font

![image](https://user-images.githubusercontent.com/89634755/132447447-793c98a1-c918-4f0f-ac51-85a1e6b99d38.png)

![image](https://user-images.githubusercontent.com/89634755/132447729-dd6a64a8-03be-48f9-9d4d-cae19a6355a7.png)

**Comment by redtoolskobe** [comment]
windows  10  

![image](https://user-images.githubusercontent.com/89634755/132452785-84cb1bd6-75b8-4b4c-9233-a3874d8b5c1d.png)

**Comment by MarvinJWendt** [comment]
Huh, that's weird. I cannot reproduce it. 

![image](https://user-images.githubusercontent.com/31022056/132647958-5701b191-0662-4a28-87b6-b1ba6710e8ce.png)


1. Did you resize the terminal afterwards?
2. What terminal are you using? (iTerm, VSC inbuilt, etc.)
3. What version of PTerm are you using?

Thanks! :)

**Comment by redtoolskobe** [comment]
use  iterm2   version 3.4.3
pterm  version  v0.12.30

**Comment by redtoolskobe** [comment]
> Huh, that's weird. I cannot reproduce it.
> 
> ![image](https://user-images.githubusercontent.com/31022056/132647958-5701b191-0662-4a28-87b6-b1ba6710e8ce.png)
> 
> 1. Did you resize the terminal afterwards?
> 2. What terminal are you using? (iTerm, VSC inbuilt, etc.)
> 3. What version of PTerm are you using?
> 
> Thanks! :)

use iterm2 version 3.4.3
pterm version v0.12.30

**Comment by MarvinJWendt** [comment]
Hmm, I am still not able to reproduce this. I tried it on several machines (macOS and Windows 10) and several terminals (iTerm, Windows Terminal, PowerShell, CMD, IntelliJ inbuilt terminal, VSC terminal,  etc.), but the result always looked like it should. 

I will leave this open for now, if anyone finds a reason for this, but since I cannot reproduce it at the moment, I don't think it's an Issue with PTerm itself.

**Comment by CodyGuo** [comment]
```shell
windows 10
go version go1.17.5 windows/amd64
github.com/pterm/pterm v0.12.33
```

code
```go
package main

import "github.com/pterm/pterm"

func main() {
	pterm.DefaultBigText.WithLetters(pterm.NewLettersFromStringWithStyle(
		"Pterm", pterm.NewStyle(pterm.FgCyan))).Render()
}
```


![image](https://user-images.githubusercontent.com/10672966/146197356-d714e0db-6c0c-400b-b41b-8d7c8576da48.png)

**Comment by zt2** [comment]
I got the same problem, changing my shell from zsh to bash and changing my terminal from iTerm2 to Terminal (with different fonts). It is always like this.
![image](https://user-images.githubusercontent.com/7644862/153056220-0ea61db6-0c57-45c2-a9c2-1d16819ae7cd.png)
Here is my debug info:
- pterm version: v0.12.36
- terminal: iTerm2 Build 3.4.15
- terminal font: JetBrains Mono
- shell: zsh
- os: macOS

**Comment by nickbug** [comment]
After my test, it maybe because of the charset problem, the following test is in windows 10 cmd.

1. cmd charset is 950, the DefaultBigText will wrong
![image](https://github.com/pterm/pterm/assets/9109840/2777ee1b-6b05-4206-8daf-401b4dd0d14f)

2. cmd charset is 65001, the DefaultBigText will wroking
![image](https://github.com/pterm/pterm/assets/9109840/d75f70bd-2af6-478b-99e8-0bda83ce35a7)

- pterm version : v0.12.63

**Comment by nickbug** [comment]
The actually problem is "go-runewidth" package which used to calculator letter width.

When init stage, this package will auto check the enviroment is east asisn or not, If yes the char "█" in letter will be see 2 width.

	func init() {
		handleEnv()
	}

	func handleEnv() {
		env := os.Getenv("RUNEWIDTH_EASTASIAN")
		if env == "" {
			EastAsianWidth = IsEastAsian()
		} else {
			EastAsianWidth = env == "1"
		}

		// update DefaultCondition
		if DefaultCondition.EastAsianWidth != EastAsianWidth {
			DefaultCondition.EastAsianWidth = EastAsianWidth
			if len(DefaultCondition.combinedLut) > 0 {
				DefaultCondition.combinedLut = DefaultCondition.combinedLut[:0]
				CreateLUT()
			}
		}
	}

So if anyone want to temporarily solve this problem, maybe can try the following two way:
(**Warning! It may be cause unexpected problem**)

1. Set environment RUNEWIDTH_EASTASIAN = 0 before startup

2. Reset the runewidth DefaultCondition in your init function

		// update DefaultCondition
		if DefaultCondition.EastAsianWidth != EastAsianWidth {
			runewidth.EastAsianWidth = false
			runewidth.DefaultCondition = runewidth.NewCondition()
			runewidth.DefaultCondition.CreateLUT()
		}

**Comment by MarvinJWendt** [comment]
Yes, looks to be an issue with `runewidth` and East Asia regions. Unfortunately, I don't know how to fix this currently. It has some open issues and PRs for years now, and I am not sure, if they would fix this behavior.

@mattn wrote this as a workaround: https://github.com/mattn/go-runewidth/issues/64#issuecomment-1221642154

There are some global variables exposed, that we could change, but I don't have an asian env set up, so I cannot test it. 

**If someone wants to test if changing one of those variables helps, please feel free to do so: https://github.com/mattn/go-runewidth/blob/44b7c5b4d67df8ca22917b6800c158a6d3be3560/runewidth.go#L13-L23**

It should also be looked for unintended side effects.

**Comment by mattn** [comment]
What is your LANG or codepage on your Windows?

**Comment by MarvinJWendt** [comment]
950 doesn't seem to work, according to this comment: https://github.com/pterm/pterm/issues/255#issuecomment-1652300342
